### ISSUE TITLE
Filterbar component

### ISSUE THREAD

**Issue by mattia-io** [OneArmyWorld/onearmy#244]
update BenGamma (25/03/19):

### Description
The **FilterBar** component will be used across different sections in the website, to helps the users search, filter and we hope, easily find what they are looking for.

### Page related
Will be used in :
- the how-to overview (#419 ) 
- the discussions overview (#402 )
- The Events section (still working on the design)
- the Map section (still working on the design)

### Mockups
* In how-to & discussion section

<img width="806" alt="Capture d’écran 2019-03-28 à 17 28 21" src="https://user-images.githubusercontent.com/4993215/55175098-f46ca200-517e-11e9-8ba8-ed3956e92f49.png">

**Full states & behavior available in the prototype :**
https://www.figma.com/proto/OZ5xbZYR4ZKNBijrs9kPJnbu/OneArmy-Platform?node-id=458%3A0&viewport=-1426%2C242%2C0.0625&scaling=scale-down-width

child of #216, related to #245

**Comment by tx4x** [comment]
hi,
possibly needless to say but as long-term PP guy, I found it always tedious if not impossible to search the forum for specifics. And as this whole thing is pretty much about research and development, I'd like to propose a slightly different take which can be beneficial in the long run

- instead of fixed and static search/display facilities, you could summarize the default search controls (category/tag filter) as predefined 'segments'. in doing so, a user might extend or add his own 'filters' as segment. 
- so a segment is nothing more than a name and a list of filter expression
- a user can get a link to such a segment and paste the link in duplicate questions for instance
- the filter editor looks like this : 

![image](https://user-images.githubusercontent.com/1205100/51977348-814cf500-2487-11e9-930e-29cc4aa9ee8c.png)

next page :

![image](https://user-images.githubusercontent.com/1205100/51977364-932e9800-2487-11e9-8501-76aecea34e45.png)

next page: filter summary 

![image](https://user-images.githubusercontent.com/1205100/51977404-ad687600-2487-11e9-8fda-2c1ccf344299.png)

after 'save as segment', the segment (filters) become available in his search bar as 'chip': 

![image](https://user-images.githubusercontent.com/1205100/51977492-e9034000-2487-11e9-9ee6-41da42531146.png)

I have done this in React, TS and material-ui, as pretty generic components. 

Anyways, such 'segments' or predefined filters come to mind :

**Shredder** 
used filters: 
tags = 'shredder'
version = v3 +
activity  = > more than 3 users
likes = > 5
must have pictures = true
must have videos = true
date range = last 6 months

**Another segment I'd love to have**
users = xxxoliver, andyn
tags = shredder, motor
type = not read
likes = > 2

anyway, do as you like :-) I am just saying, a research project should have research tools as well

**Comment by tx4x** [comment]
[here another popular pain point/issue](https://davehakkens.nl/community/forums/topic/scientifics-advises/) in the forum : searching for docs, i would offer these filters as quick filter as well:

Name : Popular documents
user points = > beginner
likes = > average
tags = selected
type = all
has attachments = true
has links [doc/pdf,...] = true

i would even pay money to have this now !! ;-)

**Comment by mattia-io** [comment]
Hey @tx4x once again thanks for all your inputs. 
Glad to have insightful tips from long time Precious Plastic-er. 

Yep, search is currently a pain. For OA we want to focus on creating well-organized content properly categorized and tagged. Rather than building a super strong search engine that would probably take forever (and realistically would probably never be as good as people has become accustomed to) we want to focus on educating our users to use the platform in a way that helps creating useful content that can be retrieved. This is also why in the new platform we won't have 'forum' as it kind of instinctively draw people to dump their thought in a chaotic manner. 

After long discussion with inhouse UX and devs we decided to go for a combo of very structure cat/tag system with a keyword search. 

Didn't fully understand the 'segment' point :(

*Thank you for the point about scientific research, very good point we haven't thought about, will bring it up in our next UX meeting. 

Very grateful for your contributions and ideas

**Comment by tx4x** [comment]
the 'segment' thing is just a pre-defined list of filters, with a name on it, placed beneath the search, as 'chip'. you could use it to setup quick search filters as admin, or user. it took me 7 days to make it, i am happy to share it, its TS, react, material-ui. i will put it as PR sometime soon when the discussion-module is a little further than now.

i am not sure it's a good idea not to give PP users the tools they need: a simple forum and instead just very narrow window of what you think as designer would be cool (it is to some degree) or worst: try to teach others. but for daily usage, research, creating ideas together it's really limiting, in top you have to maintain this kind of curated content. do as you like but I am always sad to see that PP became a fancy looking pet project only of some designers and whose limited particular imagination. no offence but it's really kinda of insult to be treated as a 12 year old in every corner of the platform , or worst: everybody needs your education. at least this what the current and the next platform also tells you right away. i mean if you assume everyone is too stupid and and needs max. limits to enter, manager or search content, then don't expect a smart or great thing to happen, simple as that :-)

anyway, browsing the posts with the pretty limited controls as you say: tag/cat isn't doing the job IMHO. you wanna reject all posts which are not fitting to your box or how does this work? you're saying no forum at all ?

I am also afraid you're focusing on the wrong user segment: impress new users with good looking stuff, instead of long-term users who have wanna do some work here. From my observation, those new users are all gone again in no time usually, trespassers basically. Too often i also saw good folks gone.

Anyway, I am glad to hear that you're at least considering some of the points. As said, I will place this 'mega search engine', somewhere, testable soon. it's just a few pages of pretty much trivial code, a filter builder basically. 

these quick filters come to mind (end users):  
- my 'friends' recent posts
- recent unread uploads
- recent unread links to documents
- i could keep going for hows on this one

these filters come to mind for house-keeping
- posts with no activity
- possibly duplicates
- recent popular threads with lots of activity

**Comment by mattia-io** [comment]
hey @tx4x the forums have always been a pain point of our platform. Super messy and hard, if not impossible, to find information user needs when they need it. We're not trying to teach people, we're just trying to build a tool that can help them connect and collaborate better.

We're not trying to impress anyone, we're trying to build a useful and usable platform built on (what we believe to be) strong UX foundations. 

Thanks for your contribution above, will dive into it and feed back as soon as I can.

Good day!

**Comment by BenGamma** [comment]
On pause waiting on redesign by the end of the week

**Comment by BenGamma** [comment]
This is on hold for now. Will reopen if appropriate in a close futur.
